# Programming for Bioinformatics
## Module 2 - Sequence Alignment

## Learning Outcomes

At the end of instruction, students will be able to:
- Tell what is an algorithm
- Write Python functions implementing algorithms
- Describe how to compare protein structures
- Describe how to compare protein sequences
- Explain what is a substitution matrix
- Describe the differences between PAM and BLOSUM matrices
- Use a substitution matrix to determine the similarity of two sequences
- Write a Python function implementing an exhaustive pairwise alignment algorithm
- Discuss dynamic programming strategies
- Implement dynamic programming algorithms in Python
- Calculate the statistical significance of a pairwise alignment
- Discuss similarity searches in databases
- Describe main differences between FASTA and BLAST

## Mid term exam
Date: Tuesday, 13 Jan 2019
Consisting in:
- One exercise or an open ended question to assess the ability to discuss substitution matrices
- A script assessing the ability to implement a basic pairwise alignment algorithm in Python

